apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -1,76 +1,98 @@
+      summary: "List package information for all projects in the repo",
       documentation:
-        'List package names, and optionally version (--version) and ' +
-        'path (--path) or full path (--full-path), for projects in the ' +
-        'current rush config.',
+        "List package names, and optionally version (--version) and " +
+        "path (--path) or full path (--full-path), for projects in the " +
+        "current rush config.",
       parser
     });
   }
 
   protected onDefineParameters(): void {
     this._version = this.defineFlagParameter({
-      parameterLongName: '--version',
-      parameterShortName: '-v',
-      description: 'If this flag is specified, the project version will be ' +
-        'displayed in a column along with the package name.'
+      parameterLongName: "--version",
+      parameterShortName: "-v",
+      description:
+        "If this flag is specified, the project version will be " +
+        "displayed in a column along with the package name."
     });
 
     this._path = this.defineFlagParameter({
-      parameterLongName: '--path',
-      parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
-        'displayed in a column along with the package name.'
+      parameterLongName: "--path",
+      parameterShortName: "-p",
+      description:
+        "If this flag is specified, the project path will be " +
+        "displayed in a column along with the package name."
     });
 
     this._fullPath = this.defineFlagParameter({
-      parameterLongName: '--full-path',
-      parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
-        'be displayed in a column along with the package name.'
+      parameterLongName: "--full-path",
+      parameterShortName: "-f",
+      description:
+        "If this flag is specified, the project full path will " +
+        "be displayed in a column along with the package name."
+    });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: "--json",
+      description: "If specified,  a json file of result will be generated."
     });
   }
 
   protected run(): Promise<void> {
     return Promise.resolve().then(() => {
-      const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
+      const allPackages: Map<string, RushConfigurationProject> = this
+        .rushConfiguration.projectsByName;
       if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
+      } else if (this._jsonFlag.value) {
+        this._serializeToJson(allPackages);
       } else {
         this._printList(allPackages);
       }
     });
   }
 
+  private _serializeToJson(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    allPackages.forEach((_config: RushConfigurationProject, name: string) => {
+      console.log(JSON.stringify(name));
```

Comment:
Instead of logging all the packages separately in a series, would it make more sense to serialize the entire dataset as an array of objects?

Also, I'm curious about something. I set `list` up to use `console.log` initially. Does anyone know if it's possible to pipe the console.log output to some other process? (Like into a json file for logging purposes?).

Reply:
Also, do you think we should include the other data (version, path, ect.) if those flags are specified in addition to --json?

Reply:
>would it make more sense to serialize the entire dataset as an array of objects?

I think that would be the expected behavior. You'd expect to be able to pipe the output of this command to a JSON parser without having to detect splits in individual JSON objects.

Reply:
>Also, do you think we should include the other data (version, path, ect.) if those flags are specified in addition to --json?

Yes, that data should be included.

I think the output with `--json` should match data in the tabular output, but in JSON.